engine/consensus/matching/engine.go

```diff
@@ -729,10 +680,11 @@ func (e *Engine) ensureStakedNodeWithRole(nodeID flow.Identifier, block *flow.He
 // seals mempool.
 func (e *Engine) sealResult(incorporatedResult *flow.IncorporatedResult) error {
 
-	// store the result to make it persistent for later checks
-	err := e.resultsDB.Store(incorporatedResult.Result)
-	if err != nil && !errors.Is(err, storage.ErrAlreadyExists) {
-		return fmt.Errorf("could not store sealing result: %w", err)
+	// at this point we expect to have receipt stored in database. Check our state before sealing.
+	// Failing here means implementation flaw.
+	_, err := e.receiptsDB.ByBlockID(incorporatedResult.Result.BlockID)
+	if err != nil {
+		return fmt.Errorf("missing receipt for execution result: %w", err)
```

Comment:
⚠️ 
You cannot use `ByBlockID` as Indexing multiple Execution Receipts for the same block is not supported atm. As consensus nodes can't populate the index you are querying here, this check will always fail. 

Even if it was implemented, the check is still incomplete: 
* finding some receipt for the block does not mean it is the receipt committing to the `incorporatedResult` 
* it could be a receipt from a different Execution Node for a different result 

Echoing Martin's and Leo's comment: I think this is best removed.